src/components/ui/Icon.tsx

```diff
@@ -1,81 +1,52 @@
-import {
-  ArrowLeftIcon,
-  BagIcon,
-  CalendarIcon,
-  ContainerIcon,
-  CrossIcon,
-  DeleteIcon,
-  FilterIcon,
-  FoodIcon_Off,
-  FoodIcon_On,
-  HomeIcon_Off,
-  HomeIcon_On,
-  LinkIcon,
-  LogoutIcon,
-  MenuCircleIcon,
-  MenuMeatballIcon,
-  PenIcon,
-  PlusIcon,
-  ProfileIcon_Off,
-  ProfileIcon_On,
-  SearchIcon,
-  TriangleDownIcon,
-  TriangleRightIcon,
-} from '@/assets/images/icons';
 import { cn } from '@/lib/tailwind/utils';
 
 import React from 'react';
 
-const icons = {
-  arrowLeft: ArrowLeftIcon,
-  bag: BagIcon,
-  calendar: CalendarIcon,
-  container: ContainerIcon,
-  close: CrossIcon,
-  delete: DeleteIcon,
-  filter: FilterIcon,
-  link: LinkIcon,
-  logout: LogoutIcon,
-  menuCircle: MenuCircleIcon,
-  menuMeatball: MenuMeatballIcon,
-  food_off: FoodIcon_Off,
-  home_off: HomeIcon_Off,
-  profile_off: ProfileIcon_Off,
-  food_on: FoodIcon_On,
-  home_on: HomeIcon_On,
-  profile_on: ProfileIcon_On,
-  pen: PenIcon,
-  plus: PlusIcon,
-  search: SearchIcon,
-  triangleDown: TriangleDownIcon,
-  triangleRight: TriangleRightIcon,
-} as const;
-
+// reference for icon size: https://tailwindcss.com/docs/width
+// Since some of the icons in the design are not following 4px grid, and to be fixed in the future,
+// If the size in the design is not in the list, you can assign the closest size from this list.
 const iconSize = {
-  sm: 'w-6 h-6',
-  md: 'w-8 h-8',
-  lg: 'w-10 h-10',
+  3: 'w-3',
+  3.5: 'w-3.5',
+  4: 'w-4',
+  5: 'w-5',
+  6: 'w-6',
+  7: 'w-7',
+  8: 'w-8',
+  9: 'w-9',
+  10: 'w-10',
+  11: 'w-11',
+  12: 'w-12',
+  13: 'w-13',
+  14: 'w-14',
+  15: 'w-15',
+  16: 'w-16',
 } as const;
 
-export type IconName = keyof typeof icons;
+const iconColor = {
+  black: 'text-black',
+  primary: 'text-primary',
+  danger: 'text-danger',
+  gray: 'text-gray',
+  'gray-dark': 'text-gray-dark',
+};
 
 export type IconSize = keyof typeof iconSize;
 
-type IconProps = {
-  iconName: IconName;
-  color?: string;
+export type IconColor = keyof typeof iconColor;
+
+type Props = {
+  icon: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
+  color?: IconColor;
   size?: IconSize;
   className?: string;
 };
 
-export const Icon: React.FC<IconProps> = ({ iconName, color, size = 'md', className }) => {
-  const IconComponent = icons[iconName];
-
-  // stroke style doesn't work if you set it stroke-colorname,
-  // but it works if you set it text-colorname
-  // and it work also without fill style
-  const colorStyle = 'text-' + color;
+export const Icon: React.FC<Props> = ({ icon, color = 'black', size = 3, className }) => {
+  const IconComponent = icon;
+  const colorStyle = iconColor[color];
   const sizeStyle = iconSize[size];
 
-  return <IconComponent className={cn(colorStyle, sizeStyle, className)} />;
+  // adding "sizeStyle" as last argument to make sure it overrides any sizing written in "className"
+  return <IconComponent className={cn(colorStyle, className, sizeStyle, 'aspect-square')} />;
```

Comment:
`className` should always come last to override the predefined tailwind classes that appear earlier since we use the `tailwind-merge`.

In this case, writing like the following would be better, I guess.
```ts
cn('aspect-square', colorStyle, sizeStyle, className)
```

Reply:
Is there any reason for not using the `cva()` variants (`class-variance-authority`) for the color and size?

Reply:
> `className` should always come last to override the predefined tailwind classes that appear earlier since we use the `tailwind-merge`.
> 
> In this case, writing like the following would be better, I guess.
> 
> ```ts
> cn('aspect-square', colorStyle, sizeStyle, className)
> ```


I thought we discussed and agreed on overriding className by "sizeStyle" and "aspect-square", to keep all the icons always square.
If it's my misunderstanding, I'll change it to order like this "cn('aspect-square', colorStyle, sizeStyle, className)" to keep it flexible, customizable with classname

Reply:
> Is there any reason for not using the `cva()` variants (`class-variance-authority`) for the color and size?

Thanks, that's a good call out!
I'll utilize cva to make it more readable & clean

Reply:
> I thought we discussed and agreed on overriding className by "sizeStyle" and "aspect-square", to keep all the icons always square.

Putting `aspect-square` at the end does not override `w-{n} h-{m}` because browsers ignore the `aspect-ratio` CSS property if both `width` and `height` properties are explicitly specified to the same element.

Reply:
I had a misunderstanding on "aspect-square", and I got it now that putting it as last one doesn't make sense.

In the first place, is "aspect-squeare" unecessary?
If we change sizeStyle property from "w-{n}" to "h-{n}", we don't need "aspect-squeare", right?

Reply:
> If we change sizeStyle property from "w-{n}" to "h-{n}", we don't need "aspect-squeare", right?

In order to make an element square using `aspect-square`, only one of `w-{n}` or `h-{n}` is necessary.
- `"aspect-square w-{n}"`
- `"aspect-square h-{n}"`

If you specify both width and height, `aspect-square` is ignored.

- `"h-4 h-6"` becomes 16*24px rectangle

This also means you can create a square without using `aspect-square` by specifying both width and height at the same size like:

- `"w-4 h-4"` becomes 16*16px square

Reply:
> If we change sizeStyle property from "w-{n}" to "h-{n}", we don't need "aspect-squeare", right?

If you prefer not to use `aspect-square`, you need to change the sizeStyle
From:
```ts
  3: 'w-3 h-3',
  3.5: 'w-3.5 h-3.5',
  4: 'w-4 h-4',
  5: 'w-5 h-5',
  6: 'w-6 h-6',
  7: 'w-7 h-7',
  8: 'w-8 h-8',
  9: 'w-9 h-9',
  10: 'w-10 h-0',
  11: 'w-11 h-1',
  12: 'w-12 h-2',
  13: 'w-13 h-3',
  14: 'w-14 h-4',
  15: 'w-15 h-5',
  16: 'w-16 h-6',
```
To:
```ts
  3: 'w-3 h-3',
  3.5: 'w-3.5 h-3.5',
  4: 'w-4 h-4',
  5: 'w-5 h-5',
  6: 'w-6 h-6',
  7: 'w-7 h-7',
  8: 'w-8 h-8',
  9: 'w-9 h-9',
  10: 'w-10 h-0',
  11: 'w-11 h-1',
  12: 'w-12 h-2',
  13: 'w-13 h-3',
  14: 'w-14 h-4',
  15: 'w-15 h-5',
  16: 'w-16 h-6',
```